src/components/sendWritable/send.js

```diff
@@ -37,24 +41,30 @@ class SendWritable extends React.Component {
         amount: {
           value: this.props.prevState.amount || this.state.amount.value,
         },
+        reference: {
+          value: this.props.prevState.reference || this.state.reference.value,
+        },
         ...authStatePrefill(this.props.account),
       };
       this.setState(newState);
     }
   }
 
-  handleChange(name, value, error) {
+  handleChange(name, required = true, value, error) {
     this.setState({
       [name]: {
         value,
-        error: typeof error === 'string' ? error : this.validateInput(name, value),
+        error: typeof error === 'string' ? error : this.validateInput(name, value, required),
       },
     });
   }
 
-  validateInput(name, value) {
-    if (!value) {
+  validateInput(name, value, required) {
+    const byteCount = encodeURI(value).split(/%..|./).length - 1;
+    if (!value && required) {
       return this.props.t('Required');
+    } else if (name === 'reference' && byteCount > 64) {
+      return this.props.t('Max 64 bytes message');
```

Comment:
I would say `Maximum length exceeded` in order not to confuse users with what bytes means